aten/src/ATen/native/cpu/DistributionTemplates.h

```diff
@@ -8,6 +8,79 @@
 
 namespace at { namespace native { namespace templates {
 
+template<typename RNG>
+static void random_from_to_kernel(TensorIterator& iter, uint64_t range, int64_t base, RNG* generator) {
+  AT_DISPATCH_ALL_TYPES_AND(at::ScalarType::Bool, iter.dtype(), "random_from_to_cpu", [&] {
+    std::lock_guard<std::mutex> lock(generator->mutex_);
+    if ((
+      std::is_same<scalar_t, int64_t>::value ||
+      std::is_same<scalar_t, double>::value ||
+      std::is_same<scalar_t, float>::value) && range >= 1ULL << 32)
+    {
+      cpu_serial_kernel(iter, [range, base, generator]() -> scalar_t {
+        return static_cast<scalar_t>(static_cast<int64_t>((generator->random64() % range) + base));
+      });
+    } else {
+      cpu_serial_kernel(iter, [range, base, generator]() -> scalar_t {
+        return static_cast<scalar_t>(static_cast<int64_t>((generator->random() % range) + base));
+      });
+    }
+  });
+}
+
+template<typename RNG>
+static void random_kernel(TensorIterator& iter, RNG* generator) {
+  std::lock_guard<std::mutex> lock(generator->mutex_);
+  if (isFloatingType(iter.dtype())) {
+    AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, iter.dtype(), "random_cpu", [&] {
+      if (std::is_same<scalar_t, double>::value) {
+        cpu_serial_kernel(iter, [generator]() -> scalar_t {
+          return generator->random64() % static_cast<uint64_t>((1ULL << std::numeric_limits<scalar_t>::digits) + 1);
+        });
+      } else {
+        cpu_serial_kernel(iter, [generator]() -> scalar_t {
+          return generator->random() % static_cast<uint64_t>((1ULL << std::numeric_limits<scalar_t>::digits) + 1);
+        });
+      }
+    });
+  } else if (isIntegralType(iter.dtype(), /*includeBool=*/true)) {
+    AT_DISPATCH_INTEGRAL_TYPES_AND(at::ScalarType::Bool, iter.dtype(), "random_cpu", [&] {
+      if (std::is_same<scalar_t, int64_t>::value) {
+        cpu_serial_kernel(iter, [generator]() -> scalar_t {
+          return generator->random64() % (static_cast<uint64_t>(std::numeric_limits<scalar_t>::max()) + 1);
+        });
+      } else if (std::is_same<scalar_t, bool>::value) {
+        cpu_serial_kernel(iter, [generator]() -> scalar_t {
+          return generator->random() & 1;
```

Comment:
Not sure, but can you please check distribution of booleans in randomly generated tensor